src/SFA.DAS.Assessor.Functions/Domain/Print/PrintCommand.cs

```diff
@@ -0,0 +1,128 @@
+using SFA.DAS.Assessor.Functions.ExternalApis.Assessor.Constants;
+using SFA.DAS.Assessor.Functions.Infrastructure;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.Assessor.Functions.Domain.Print
+{
+    public class PrintCommand : IPrintCommand
+    {
+        private readonly ILogger<PrintCommand> _logger;
+        private readonly IPrintCreator _printCreator;
+        private readonly IBatchService _batchService;
+        private readonly IScheduleService _scheduleService;
+        
+        private readonly INotificationService _notificationService;
+        private readonly IFileTransferClient _externalFileTransferClient;
+        private readonly IFileTransferClient _internalFileTransferClient;
+        private readonly CertificatePrintFunctionSettings _settings;
+
+        public PrintCommand(
+            ILogger<PrintCommand> logger,
+            IPrintCreator printCreator,
+            IBatchService batchService,
+            IScheduleService scheduleService,
+            INotificationService notificationService,
+            IFileTransferClient externalFileTransferClient,
+            IFileTransferClient internalFileTransferClient,
+            IOptions<CertificatePrintFunctionSettings> options)
+        {
+            _logger = logger;
+            _printCreator = printCreator;
+            _batchService = batchService;
+            _scheduleService = scheduleService;
+            _notificationService = notificationService;
+            _externalFileTransferClient = externalFileTransferClient;
+            _internalFileTransferClient = internalFileTransferClient;
+            _settings = options?.Value;
+
+            _externalFileTransferClient.ContainerName = _settings.PrintRequestExternalBlobContainer;
+            _internalFileTransferClient.ContainerName = _settings.PrintRequestInternalBlobContainer;
+        }
+
+        public ICollector<string> StorageQueue { get; set; }
+
+        public async Task Execute()
+        {
+            try
+            {
+                _logger.Log(LogLevel.Information, "Print command started");
+
+                var schedule = await _scheduleService.Get();
+                if (schedule == null)
+                {
+                    _logger.Log(LogLevel.Information, "There is no print schedule which allows printing at this time");
+                    return;
+                }
+
+                var nextPrintBatchNumber = await _batchService.BuildPrintBatchReadyToPrint(schedule.RunTime, _settings.AddReadyToPrintChunkSize);
+                if (nextPrintBatchNumber != null)
+                {
+                    var certificates = (await _batchService.GetCertificatesForBatchNumber(nextPrintBatchNumber.Value)).Sanitise(_logger);
+                    if (certificates.Count > 0)
+                    {
+                        var batch = await _batchService.Get(nextPrintBatchNumber.Value);
```

Comment:
do we need an entire other network call for the created Date of the batch? We already have called to create the batch / get the batch number and then get the certificates for that batch number.

Reply:
Probably it could be reworked, however the batch is calling the batch controller and the certificates the certificate controller, so this would be a lot of rework to join them together, for a very small network gain, so I would rather leave it how it is